portal-kernel/src/com/liferay/document/library/kernel/store/StoreArea.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.kernel.store;
+
+import com.liferay.petra.function.UnsafeRunnable;
+import com.liferay.petra.lang.CentralizedThreadLocal;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+/**
+ * @author Adolfo Pérez
+ */
+public enum StoreArea {
+
+	DELETED("_deleted"), EVICTED("_evicted"), LIVE(StringPool.BLANK);
+
+	public static String getPath(String... path) {
+		StoreArea storeArea = _storeAreaThreadLocal.get();
+
+		return storeArea._namespace + StringPool.SLASH +
+			StringUtil.merge(path, StringPool.SLASH);
+	}
+
+	public static <T extends Throwable> void withStoreArea(
+			StoreArea readStoreArea, UnsafeRunnable<T> unsafeRunnable)
```

Comment:
Yes, that was a mistake. Originally I had two active areas (one for reading, one for writting) because I was thinking about adding a "move" operation to the Store interface (the way things are moved right now transfers data unnecessarily).